plugins/woocommerce-admin/client/homescreen/layout.js

```diff
@@ -76,9 +77,11 @@ export const Layout = ( {
 		( userPrefs.homepage_layout || defaultHomescreenLayout ) ===
 			'two_columns' && hasTwoColumnContent;
 
-	if ( isBatchUpdating && ! showInbox ) {
-		setShowInbox( true );
-	}
+	useEffect( () => {
+		if ( isBatchUpdating && ! showInbox ) {
+			setShowInbox( true );
+		}
+	}, [ isBatchUpdating, showInbox ] );
```

Comment:
`setState` should not occur outside of an effect.

Reply:
I don't think this is a general rule, right?

Unless I'm missing something, I don't even see where `showInbox` is used at all! So, I think we can just totally remove this bit of state.

Reply:
Good catch, @mattsherman! It seems the `showInbox` usage was removed in [this PR](https://github.com/woocommerce/woocommerce-admin/pull/5465/files#diff-de1bf1e6e26b24ec1623b1a6c816a8fb3984b05375c320c8ad82d44701f59ad7L85).

Reply:
Removed this in [1fd1e86](https://github.com/woocommerce/woocommerce/pull/37641/commits/1fd1e8638ab613effcba9889fd503a7b49669eb1)

Reply:
`setState` outside an effect, directly during render, can be fine in some cases, and this one seems to have been one of them. When the `isBatchUpdating` prop changed to `true`, it was syncing the `showInbox` state to be `true`, too. Kind of a hook version of `getDerivedStateFromProps`.

When `setState` is called directly in render, then the result of the current render is guaranteed to be never used. Not committed to DOM, no effects running. Because the render was kind of inconsistent: `isBatchUpdating` is already `true`, but `showInbox` can still be `false`. Instead, the `setState` is scheduled to be run as soon as possible, and triggers a new render, with the new, consistent, state values.

`getDerivedStateFromProps`, internally, has the same semantics and scheduling.